### ISSUE TITLE
Feature request: Support drag-and-drop reordering for terminal tabs

### ISSUE THREAD

**Issue by htazq** [GOODBOY008/r-shell#16]
It would be helpful if terminal tabs could be reordered by drag and drop. Currently, after opening multiple SSH sessions, the tab order appears to be fixed, and users cannot freely drag an active tab to another position. When working with many servers or grouping related sessions, being able to rearrange tabs manually would make the workflow much more convenient. Please consider adding drag-and-drop tab reordering, similar to common terminal clients, browsers, or tools like MobaXterm.

**Comment by GOODBOY008** [comment]
Impled in main via commit：[397e8cbb8dd77fa95f7d8d9b0cb3ab080ccc1427](https://github.com/GOODBOY008/r-shell/commit/397e8cbb8dd77fa95f7d8d9b0cb3ab080ccc1427)
